lib/grape/dsl/inside_route.rb

```diff
@@ -26,42 +26,56 @@ def self.post_filter_methods(type)
       # Methods which should not be available in filters until the before filter
       # has completed
       module PostBeforeFilter
-        def declared(params, options = {}, declared_params = nil)
+        def declared(passed_params, options = {}, declared_params = nil)
           options = options.reverse_merge(include_missing: true, include_parent_namespaces: true)
+          declared_params ||= optioned_declared_params(options)
 
-          # Declared params including parent namespaces
-          all_declared_params = route_setting(:saved_declared_params).flatten | Array(route_setting(:declared_params))
-
-          # Declared params at current namespace
-          current_namespace_declared_params = route_setting(:saved_declared_params).last & Array(route_setting(:declared_params))
-
-          declared_params ||= options[:include_parent_namespaces] ? all_declared_params : current_namespace_declared_params
-
-          raise ArgumentError, 'Tried to filter for declared parameters but none exist.' unless declared_params
-
-          if params.is_a? Array
-            params.map do |param|
-              declared(param || {}, options, declared_params)
-            end
+          if passed_params.is_a?(Array)
+            declared_array(passed_params, options, declared_params)
           else
-            declared_params.each_with_object(Hashie::Mash.new) do |key, hash|
-              key = { key => nil } unless key.is_a? Hash
+            declared_hash(passed_params, options, declared_params)
+          end
+        end
 
-              key.each_pair do |parent, children|
-                output_key = options[:stringify] ? parent.to_s : parent.to_sym
+        def declared_array(passed_params, options, declared_params)
+          passed_params.map do |passed_param|
+            declared(passed_param || {}, options, declared_params)
+          end
+        end
 
-                next unless options[:include_missing] || params.key?(parent)
+        def declared_hash(passed_params, options, declared_params)
+          declared_params.each_with_object(Hashie::Mash.new) do |declared_param, memo|
+            # if it is not a Hash then it does not have children.
```

Comment:
Nitpick, this comment should not have a period if it's not a full sentence starting with a Capital.